components/script/script_thread.rs

```diff
@@ -3828,6 +3831,20 @@ impl ScriptThread {
         context.process_response_eof(Ok(ResourceFetchTiming::new(ResourceTimingType::None)));
     }
 
+    // refer definition of pre_page_load method in "script_thread.rs"; code taken from "start_page_load_about_blank"
+    fn page_load_about_srcdoc(&self, load_data: LoadData) {
```

Comment:
You will likely end up with a method that is 90% similar to start_page_load_about_blank. We will use the srcdoc string as the initial content for the parser (in process_response_chunk), and we'll use about:srcdoc instead of about:blank, but otherwise I suspect the rest can be used unchanged.

Reply:
Modified this method. Please review it.